src/org/zaproxy/zap/extension/ascanrulesBeta/InsecureHTTPMethod.java

```diff
@@ -1,455 +1,702 @@
+                                        // highlighted
+                                        // in
+                                        // the
+                                        // message
+                                        alertMessage);
+                            }
+                        } else {
+                            if (log.isDebugEnabled()) {
+                                log.debug(enabledmethod + "!=" + insecureMethod);
+                            }
+                        }
+                    }
+                }
+            }
+        } catch (Exception e) {
+            log.error("Error scanning a Host for Insecure HTTP Methods: " + e.getMessage(), e);
+        }
+
+    }
+
+    @Override
+    public int getRisk() {
+
+        return Alert.RISK_MEDIUM;
+    }
+
+    @Override
+    public int getCweId() {
+
+        return 200; // Information Exposure (primarily via TRACK / TRACE)
+    }
+
+    @Override
+    public int getWascId() {
+
+        return 45; // Fingerprinting
+    }
+
+    private class MessageAndEvidence {
+
+        public HttpMessage message;
+
+        public String evidence;
+
+        public MessageAndEvidence(HttpMessage message, String evidence) {
+            this.message = message;
+            this.evidence = evidence;
+        }
+    }
+
+    private class DescriptionAndExtraInfo {
+
+        public String description;
+
+        public String extraInfo;
+
+        public DescriptionAndExtraInfo() {
+        }
+
+        @SuppressWarnings("unused")
+        public DescriptionAndExtraInfo(String description, String extraInfo) {
+            this.description = description;
+            this.extraInfo = extraInfo;
+        }
+    }
+
+
+    private MessageAndEvidence testTraceOrTrack(HttpMessage baseMsg, String method) throws Exception {
+
+        HttpRequestHeader traceRequestHeader = baseMsg.getRequestHeader();
+        traceRequestHeader.setMethod(method);
+        // TRACE is supported in 1.0. TRACK is presumably the same, since it is
+        // a alias for TRACE. Typical Microsoft.
+        traceRequestHeader.setVersion(HttpRequestHeader.HTTP10);
+
+        HttpMessage tracemsg = getNewMsg();
+        tracemsg.setRequestHeader(traceRequestHeader);
+        String randomcookiename = RandomStringUtils.random(15, "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789");
+        String randomcookievalue = RandomStringUtils.random(40, "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789");
+        TreeSet<HtmlParameter> cookies = tracemsg.getCookieParams();
+        cookies.add(new HtmlParameter(HtmlParameter.Type.cookie, randomcookiename, randomcookievalue));
+        tracemsg.setCookieParams(cookies);
+
+        sendAndReceive(tracemsg, false); // do not follow redirects. That might
+        // ruin our day.
+        // if the response *body* from the TRACE request contains the cookie,
+        // we're in business :)

```

Comment:
All of these 4 comment lines. Could be a block comment ahead of the `sendAndReceive`